x-pack/plugins/infra/public/components/log_stream/README.md

```diff
@@ -0,0 +1,73 @@
+# Embeddable `<LogStream />` component
+
+The purpose of this component is to allow you, the developer, to have your very own Log Stream in your plugin.
+
+The plugin is exposed through `infra/public`. Since Kibana uses relative paths is up to you to find how to import it (sorry).
+
+```tsx
+import { LogStream } from '../../../../../../infra/public';
+```
+
+## Prerequisites
+
+To use the component, there are several things you need to ensure in your plugin:
+
+- In your `kibana.json` plugin, you need to either add `"requiredBundles": ["infra"]` or `"requiredPlugins": ["infra"]`.
```

Comment:
Mmm, good point. I haven't tried what would happen if the infra plugin is disabled.

Reply:
Maybe we can provide a "is the logs api available?" wrapper alongside that renders an informative message instead of the `children`? (In a separate PR, though.)

Reply:
I checked what happens with `xpack.infra.enabled: false`. The component renders, but it says there's no log data. I will remove the `requiredBundles` bit from the docs because it's a lie :D.

Reply:
I think I want to give freedom to plugin developers here. They can either choose to make it a hard requirement (`requiredPlugins`), or a soft one (`optionalPlugins`). In the second case, we can advise to wrap our component in a check that tests if the infra plugin is loaded or not, alongside other changes that they want to do in their UI.

```tsx
function SomeComponent() {
	const { services } = useKibana();
	const hasInfraPlugin = 'infra' in services; // This seems to work

	const tabs = ['Timeline', 'Metadata'];
	if (hasInfraPlugin) {
		tabs.push('Logs');
	}
	// ...
}
```

We can give examples in the docs. Do you think this is enough, or do you think it's the responsibility of our component to check for this? We can of course _also_ check on our side and raise an exception.

Reply:
I think if it's in optional plugins, then they need to add it to requiredBundles as I understand it.